### PR TITLE
FEATURE: Add swipe left home screen shortcuts

### PR DESCRIPTION
**Pull request by pmusaraj**
Includes:

- a notch element for easier site dismissal
- a new swipe button for the chat feature
- a new swipe button for the hot feature
- a new swipe button for "take me back to my last spot"
- for unauthenticated sites, a swipe right button for "Connect"
- adds a time for the main bar "Connect" button, hides it after 3 days of site being added to home screen (see https://meta.discourse.org/t/read-only-subscription-forums-on-discoursehub-and-connect-button/325266)
- removal of `supportsDelegatedAuth` logic (been in core for years now)
- moves WebViewComponent to its own file (in prep for other changes)
- some dependency updates

### CODE REVIEW

File: js/Discourse.js [hunk 1]
```diff
@@ -234,13 +234,8 @@ class Discourse extends React.Component {
...
 
     if (url) {
-      this._siteManager.setActiveSite(url).then(activeSite => {
-        let supportsDelegatedAuth = false;
-        if (this._siteManager.supportsDelegatedAuth(activeSite)) {
-          supportsDelegatedAuth = true;
-        }
-        this.openUrl(url, supportsDelegatedAuth);
-      });
+      this._siteManager.setActiveSite(url);
```
Comment (jjaffeux) [inline js/Discourse.js:237]: So this is not a promise anymore?

File: js/Discourse.js [hunk 2]
```diff
@@ -329,7 +321,7 @@ class Discourse extends React.Component {
         sound: true,
       });
 
-      addShortcutListener(({userInfo, activityType}) => {
+      addShortcutListener(({userInfo, _activityType}) => {
```
Comment (jjaffeux) [inline js/Discourse.js:324]: why do we keep this param then if we dont use it?

**Review by jjaffeux — APPROVED** [review]

File: js/Discourse.js (hunk 1 above)
Comment (pmusaraj) [reply js/Discourse.js:237]: Right, we no longer need to wait for the promise to complete so we can find out if it supports delegated auth. Delegated auth was added 3+ years ago, we now assume all Discourse sites have support for it.

File: js/Discourse.js (hunk 2 above)
Comment (pmusaraj) [reply js/Discourse.js:324]: It's not needed indeed.

**Pull request merged**